dashboard/src/components/workspace/WorkspaceInvoices.tsx

```diff
@@ -6,37 +6,37 @@ import { triggerToast } from '@/utils/toast';
 import { useState } from 'react';
 
 export function WorkspaceInvoices() {
+  const { currentWorkspace } = useCurrentWorkspaceAndApplication();
   const [loading, setLoading] = useState(false);
 
-  const { currentWorkspace } = useCurrentWorkspaceAndApplication();
+  const handleViewInvoices = async () => {
+    setLoading(true);
```

Comment:
A hook that would manage loading / error states would be so great for functions 👀